### PR TITLE
[EMB-562] Add editable license field

### PR DESCRIPTION
**Pull request by fabmiz**
## Purpose

Add editable license field

## Summary of Changes

- Rename `registry-provider` to `registration-provider`
- Add `<RegistriesLicensePicker>` component
- Update mirage setup
- Fix `<LicenseText>` component to display to wrap text correctly.

## Feature Flags

`ember_registries_detail_page`

## Ticket

[[EMB-562]](https://openscience.atlassian.net/browse/EMB-562)

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: lib/registries/addon/components/registries-license-picker/template.hbs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                        @rel='noopener'
+                        @href={{this.helpLink}}
+                    >
+                        {{t 'app_components.license_picker.faq'}}
+                    </OsfLink>
+                </div>
+
+                {{#if this.selectedLicense}}
+                    {{#if this.requiredFields}}
+                        <br>
```
Comment (aaxelb) [inline lib/registries/addon/components/registries-license-picker/template.hbs]: We generally shouldn't use `<br>` for formatting. This looks like something better served by styling on the custom input below?

File: lib/registries/addon/components/registries-license-picker/template.hbs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    {{/if}}
+
+                    <form.custom @valuePath='nodeLicense'>
+                        {{#each this.requiredFields as |key|}}
+                            <br>
+                            <label for='nodeLicense-{{key}}'>{{t (concat 'app_components.license_picker.fields.' key)}}</label>
```
Comment (aaxelb) [inline lib/registries/addon/components/registries-license-picker/template.hbs]: Minor: Instead of building ids by hand (and having to keep them in sync), I think it's cleaner to put the input inside the label, whenever possible.

File: lib/registries/addon/components/registries-license-picker/styles.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    .Save {
+        margin-right: 15px;
+        font-weight: 600;
+        margin-right: 10px;
+        border: 0;
+
+        &:hover,
+        &:active,
+        &:focus {
+            background-color: $secondary-blue;
```
Comment (aaxelb) [inline lib/registries/addon/components/registries-license-picker/styles.scss]: Avoid overriding OsfButton styles

File: lib/registries/addon/components/registries-license-picker/component.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+        if (name) {
+            yield timeout(500);
+        }
+
+        const licensesAcceptable = this.providerLicenses
+            .filter(license => license.get('name').includes(name || ''));
+
+        this.setProperties({ licensesAcceptable });
+        this.registration.notifyPropertyChange('license');
```
Comment (aaxelb) [inline lib/registries/addon/components/registries-license-picker/component.ts]: What is this for? It seems like code smell -- if logic elsewhere depends on this `notifyPropertyChange`, we're probably doing something wrong.

File: lib/registries/addon/components/registries-license-picker/component.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    @action
+    changeLicense(this: RegistriesLicensePicker, selected: License) {
+        this.set('selectedLicense', selected);
+        this.registration.setNodeLicenseDefaults(selected.requiredFields);
+    }
+
+    @action
+    notify(this: RegistriesLicensePicker) {
+        this.registration.notifyPropertyChange('nodeLicense');
```
Comment (aaxelb) [inline lib/registries/addon/components/registries-license-picker/component.ts]: Ditto above, why is this needed and is there a better way? I'd consider `notifyPropertyChange` an anti-pattern.

File: lib/registries/addon/components/registries-license-picker/component.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        const licensesAcceptable = this.providerLicenses
+            .filter(license => license.get('name').includes(name || ''));
+
+        this.setProperties({ licensesAcceptable });
+        this.registration.notifyPropertyChange('license');
+
+        return licensesAcceptable;
+    }).on('didReceiveAttrs').restartable(),
```
[the remark this answers is not in the document]
Comment (fabmiz) [reply lib/registries/addon/components/registries-license-picker/component.ts]: So the nice thing here is that, when the user clicks the select trigger they will see a list with all provider licenses and then they can type in a name or part of it to filter through the list. It can be done in a better way though.

File: lib/registries/addon/components/registries-license-picker/template.hbs [hunk 7]
```diff
@@ -0,0 +1,91 @@
+{{#if @manager.inEditMode}}
+    <ValidatedModelForm
```
Comment (aaxelb) [inline lib/registries/addon/components/registries-license-picker/template.hbs]: Hmm, the manager pattern doesn't mesh well with `ValidatedModelForm`, since VMF itself manages saving changes. We should probably break it into pieces, like a "smart" `ValidatedChangeManager` and a collection of "dumb" rendering components.

Talked in person -- this approach is fine for now, but we should be thinking about how our form handling can be improved.

**Comment by fabmiz** [comment]
@aaxelb fixed!
